File: demo/app/src/main/java/brisksoft/com/ad340/DetailActivity.java
package brisksoft.com.ad340;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.util.Log;
import android.widget.TextView;

public class DetailActivity extends AppCompatActivity {
    private static final String TAG = DetailActivity.class.getSimpleName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        // Get the Intent that started this activity and extract the string
        Intent intent = getIntent();
        String message = intent.getStringExtra(MainActivity.EXTRA_MESSAGE);

        // Capture the layout's TextView and set the string as its text
        TextView textView = (TextView) findViewById(R.id.textView);
        textView.setText(message);
    }

    @Override
    public void onStart() {
        super.onStart();  // Always call the superclass method first
        Log.d(TAG, "onStart()");
    }

    @Override
    public void onResume() {
        super.onResume();  // Always call the superclass method first
        Log.d(TAG, "onResume()");
    }

    @Override
    public void onPause() {
        super.onPause();  // Always call the superclass method first
        Log.d(TAG, "onPause()");
    }
    @Override
    public void onStop() {
        super.onStop();  // Always call the superclass method first
        Log.d(TAG, "onStop()");
    }
    @Override
    public void onDestroy() {
        super.onDestroy();  // Always call the superclass method first
        Log.d(TAG, "onDestroy()");
    }
}

File: demo/app/src/main/java/brisksoft/com/ad340/DetailActivity.kt
package brisksoft.com.ad340

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import kotlinx.android.synthetic.main.activity_detail.*
import kotlinx.android.synthetic.main.toolbar.*

class DetailActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)

        setSupportActionBar(toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)

        // Get the Intent that started this activity and extract the string
        val message = intent.getStringExtra(MainActivity.EXTRA_MESSAGE)

        // Capture the layout's TextView and set the string as its text
        textView.text = message
    }

    public override fun onStart() {
        super.onStart()  // Always call the superclass method first
        Log.d(TAG, "onStart()")
    }

    public override fun onResume() {
        super.onResume()  // Always call the superclass method first
        Log.d(TAG, "onResume()")
    }

    public override fun onPause() {
        super.onPause()  // Always call the superclass method first
        Log.d(TAG, "onPause()")
    }

    public override fun onStop() {
        super.onStop()  // Always call the superclass method first
        Log.d(TAG, "onStop()")
    }

    public override fun onDestroy() {
        super.onDestroy()  // Always call the superclass method first
        Log.d(TAG, "onDestroy()")
    }

    companion object {
        private val TAG = DetailActivity::class.java!!.getSimpleName()
    }
}
